R/te_datastore_duckdb.R

```diff
@@ -79,17 +79,22 @@ setMethod(
   definition = function(object, period, subset_condition) {
     checkmate::assert_integerish(period, null.ok = TRUE, any.missing = FALSE, lower = 0)
     if (use_subset <- !is.null(subset_condition)) {
-      subset_expr <- str2lang(subset_condition)
-    }
-    query <- if (is.null(period)) {
-      "SELECT * FROM trial_data"
-    } else {
-      paste0("SELECT * FROM trial_data WHERE trial_period IN (", paste0(period, collapse = ", "), ")")
+      subset_expr <- translate_to_sql(subset_condition)
     }
+    q_p1 <- "SELECT * FROM trial_data"
+    q_p2 <- ""
+    if (!is.null(period) | use_subset) q_p2 <- " WHERE"
```

Comment:
Maybe something like
```suggestion
    q_p2 <- if (!is.null(period) | use_subset) " WHERE" else ""
```
or with `ifelse(!is.null(period) | use_subset), " WHERE", "")`

Reply:
and the same for the other query parts